### PR TITLE
edited model

### PR DESCRIPTION
**Pull request by NMandel527**

### CODE REVIEW

File: src/main/java/bbm/brickbreaker/Ball.java [hunk 1]
```diff
@@ -7,27 +7,60 @@ public class Ball extends Ellipse2D.Double {
...
         return x < width && x > 0 && y < height && y > 0;
     }
-}
+
+    public boolean hitsWall(double radius) {
+        return (x - radius <= 0 || x + radius >= width
+                || y - radius <= 0 || y + radius >= height);
+    }
+
+}
```
Comment (AlizaBuchen) [inline src/main/java/bbm/brickbreaker/Ball.java:67]: add a getWidth() and getHeight() method

File: src/main/java/bbm/brickbreaker/Ball.java [hunk 2]
```diff
@@ -7,27 +7,60 @@ public class Ball extends Ellipse2D.Double {
     private int angle;
     private final int width;
     private final int height;
+    private double x;
+    private double y;
```
Comment (cblitz2) [inline src/main/java/bbm/brickbreaker/Ball.java:12]: add rows and cols

File: src/main/java/bbm/brickbreaker/Brick.java [hunk 3]
```diff
@@ -7,6 +7,8 @@ public class Brick extends Rectangle {
     private final Random rand = new Random();
     private final int width;
     private final int height;
```
Comment (AlizaBuchen) [inline src/main/java/bbm/brickbreaker/Brick.java:9]: TODO: make rows and cols not hardcoded

**Review by AlizaBuchen — APPROVED** [review]

**Review by cblitz2 — APPROVED** [review]

File: src/main/java/bbm/brickbreaker/Ball.java [hunk 4]
```diff
@@ -7,27 +7,61 @@ public class Ball extends Ellipse2D.Double {
...
 
     public double locationX() {
-        return getX() + Math.cos(angle) * VELOCITY;
+        return getX() + Math.cos(Math.toRadians(angle)) * VELOCITY;
     }
 
     public double locationY() {
-        return getY() + Math.sin(angle) * VELOCITY;
+        return getY() + Math.sin(Math.toRadians(angle)) * VELOCITY;
     }
```
Comment (andrewoid) [inline src/main/java/bbm/brickbreaker/Ball.java:40]: What do these methods do?

Comment (NMandel527) [reply src/main/java/bbm/brickbreaker/Ball.java:40]: It converts the angle to radians so it can properly calculate the sine and cosine of the angles

Comment (NMandel527) [reply src/main/java/bbm/brickbreaker/Ball.java:40]: They get the x and y locations of the ball based on the current x and y, the direction the ball is facing, and the velocity at which the ball is moving.

Comment (andrewoid) [reply src/main/java/bbm/brickbreaker/Ball.java:40]: Deserves a better method name then

**Pull request merged**